Add node degree subsection (#7)
greenelab/xswap-manuscript: content/02.body.md

## Introduction

> Networks contain information about relationships between entities ("edges between nodes").

Reviewer: I think that this first paragraph of the introduction should be the last paragraph of the introduction, and then it should get revised to fit there. Otherwise I really like this intro!

## Introduction

Most biomedical data networks are imperfect representations of the true set of relationships.
Real networks often mistakenly include edges that do not exist and exclude edges that do exist.
How well a network represents the true relationships it attempts to represent depends on a number of factors, especially the methods used to generate the data in the network [@doi:10.1016/j.jprot.2014.01.020; @doi:10.3389/fgene.2015.00260; @doi:10.1038/nbt1116].

> We define "degree bias" as the type of misrepresentation that occurs when the fraction of incorrectly existent/nonexistent relationships depends on the number of connections that nodes make (their "degrees").

Reviewer: What is the link (is there one?) between inspection bias and degree bias?

Author: I intended to say that inspection bias is a phenomenon that can cause degree bias. I hope my phrasing change in the sentence that follows helps to clarify this point.

Inspection bias indicates that entities are not uniformly studied [@doi:10.1038/nature04209], and it is likely to cause degree bias when networks are constructed using hypothesis-driven findings extracted from the literature, as newly-discovered relationships are not randomly sampled from the set of all true relationships.

> For example, the number of publications mentioning a gene has little correlation with its degree in a systematically-derived protein interaction network [Figure 6A in @doi:10.1016/j.cell.2014.10.050].

Reviewer: I think the sentence before this should be part of this paragraph.

Therefore, the high correlation between number of publications and degree in low-throughput interaction networks is almost entirely the result of inspection bias.
This evidence suggests that many poorly studied genes have similar numbers of interactions as the genes scientists have preferentially examined.

> Depending on the type of data being represented, degree biases can also arise due to experimental methods or other factors [@doi:10.1016/j.jprot.2014.01.020].

Reviewer: This feels like maybe it belongs immediately after the sentence defining degree bias.

The consequence of degree bias is that a difference in degree between two nodes may not reflect a difference in the number of true edges.
Reliance on degree may be unfavorable depending on the prediction/analysis task being conducted and the magnitude of possible degree bias in the data.

> Another reason why a method's reliance on degree can be unfavorable is that degree imbalance can lead to prediction nonspecificity.

Reviewer: But then I'm not quite sure what to do with this.

  Shown are seven degree distributions for six edge type subnetworks of Hetionet [@doi:10.7554/eLife.26726].
  (The final two are the source and target distributions of the gene-regulates-gene directed edge type.)
](https://github.com/greenelab/xswap-analysis/raw/5a257c9d28e0a4c3cc8f2fd8493f27ff4652f2c6/img/hetionet_degrees.png){#fig:hetionet width="100%"}

Consequently, degree-based predictions are more likely nonspecific, meaning the same set of predictions performs well for different tasks.
However, depending on the prediction task, edge predictions between very high degree nodes may be undesired, uninsightful, or nonspecific.
Model evaluation is challenging in this context, as nonspecific or trivial predictions need not be incorrect, even if they are not the desired outputs of the predictive model.
For example, predicting that the highest degree node in a network shares edges with the remaining nodes to which it is not connected will often lead to many correct predictions, despite the predictions being generic to all nodes in the network.

> To clarify the situation, degree-based features must be included in the interpretation of predictions in order to disentangle desired from non-desired effects and to more effectively evaluate and compare predictive models.

Reviewer: ```suggestion
Degree-based features should often be included in the interpretation of predictions to disentangle desired from non-desired effects and to effectively evaluate and compare predictive models.
```

> Degree is an important metric for differentiating between nodes.

Reviewer: I think that you could remove this and the three following sentences. Perhaps they fit somewhere else in the intro there.

### Node degree

Networks contain information about relationships between entities ("edges between nodes").
One key aspect of a node (especially in unweighted networks) is its degree, the number of connections a node makes with other nodes in the network.

> Degree is an important metric for differentiating between nodes and it is heavily relied-upon by a number of edge prediction (also known as "link prediction") features (Figure {@fig:degree}E).

Reviewer 2: ```suggestion
Degree is an important metric for differentiating between nodes.
Edge prediction (also known as "link prediction") methods tend to be heavily influenced by degree (Figure {@fig:degree}E).
```

In this paper, we introduce a permutation-based framework to find edge existence probabilities due to node degree and to quantify the contribution of degree to edge prediction methods.

Most biomedical data networks are imperfect representations of the true set of relationships.
Real networks often mistakenly include edges that do not exist and exclude edges that do exist.
How well a network represents the true relationships it attempts to represent depends on a number of factors, especially the methods used to generate the data in the network [@doi:10.1016/j.jprot.2014.01.020; @doi:10.3389/fgene.2015.00260; @doi:10.1038/nbt1116].
We define "degree bias" as the type of misrepresentation that occurs when the fraction of incorrectly existent/nonexistent relationships depends on degree.
Inspection bias indicates that entities are not uniformly studied [@doi:10.1038/nature04209], and it is likely to cause degree bias when networks are constructed using hypothesis-driven findings extracted from the literature, as newly-discovered relationships are not randomly sampled from the set of all true relationships.

> The presence of inspection bias in a network means that well-studied entities have a higher fraction of their true relationships represented than less well-studied entities.

Reviewer 2: Figure 4A at https://doi.org/10.1016/j.cell.2014.10.050 is a brilliant example of inspection bias.

> (A) Adjacency matrices showing Lit-BM-13 (blue) and HI-II-14 (purple) interactions, with proteins in bins of ∼350 and ordered by number of publications along both axes. The color intensity of each square reflects the total number of interactions for the corresponding bins.
>
> ![](https://ars.els-cdn.com/content/image/1-s2.0-S0092867414014226-gr4.jpg)

We could cite it here or add a sentence:

> For example, the number of publications mentioning a gene has little correlation with its degree in a _systematically-derived_ protein interaction network [Figure 6A in @doi:10.1016/j.cell.2014.10.050]. Therefore, the high correlation between number of publications and degree in low-throughput interaction networks is almost entirely the result of inspection bias. In reality, poorly studied genes have just as many interactions as the genes scientists have preferentially examined.

Author: Awesome suggestion!

Figure {@fig:degree}D shows examples of different degree distributions in six biomedical networks.
Degree imbalance leads high-degree nodes to dominate in the predictions made by degree-associated methods [@doi:10.1093/bioinformatics/btv215], which have been shown to be effective predictors of connections in some biological networks [@doi:10.1186/1752-0509-2-11].
Consequently, degree-based predictions are more likely nonspecific, meaning the same set of predictions would appear to perform well for different tasks.
However, depending on the prediction task and the desired types of predictions, edge predictions between very high degree nodes may be uninsightful or nonspecific.
Model evaluation can also be complicated, as nonspecific or trivial predictions need not be incorrect, even if they are not the desired outputs of the predictive model.
Rather, degree-based features must be included in the interpretation of predictions in order to disentangle desired from non-desired effects and to more effectively evaluate and compare predictive models.

![**A.** Degree distributions of networks with and without degree bias can be very different.

>   Protein-protein interaction (PPI) literature-derived network from STRING (Szklarczyk et al.) [@doi:10.1093/nar/gky1131], high-throughput-derived networks from Rual et al. [@doi:10.1038/nature04209] and Rolland et al. [@doi:10.1016/j.cell.2014.10.050].

Reviewer 2: No need for `Szklarczyk et al.` when you cite the publication and name the RESOURCE.

  Transcription factor-target gene (TF-TG) literature-derived network from Han et al. [@doi:10.1093/nar/gkx1013] and high-throughput network from Lachmann et al. [@doi:10.1093/bioinformatics/btq466].
  BioRxiv data from Rxivist [@doi:10.1101/515643; @doi:10.5281/zenodo.2566421] had edges split between those existing before and during/after 2018.
  **B.** Uniform random sampling edges produces node degrees that are linearly correlated with those in the unsampled network.
  Shown are the same PPI and TF-TG networks as in subfigure A.
  70% of literature edges were sampled with uniform probability.
  **C.** Node degree in a literature-derived network is not linearly correlated with degree in a systematically-derived network.
  Assuming that the inclusion of an edge in the systematic network is independent of node degree, the systematic degree should be linearly correlated with the true number of relationships.
  Edges found in literature-derived networks represent a non-random sample from systematically-derived edges.
  Shown are the same PPI and TF-TG networks as in subfigure A.

>   **D.** Degree distribution can vary greatly between networks, even between the source and target degree distributions within the same network.

Reviewer 2: Panel D: best if you can use full name of edge and mention in caption that degree is of the first node. Then do like

+ Anatomy→expresses→Gene
+ Anatomy→upgregulates→Gene
+ Compound–resembles–Compound
+ Disease→associates→Gene
+ Disease→downregulates→Gene
+ Gene→resembles→Gene
+ Gene←resembles←Gene

Can we fit this? The abbreviations are not good for reader comprehension

Author: Agree! Just to be sure, `G<rG` means "Gene←Downregulates←Gene", not "Gene←resembles←Gene", right?

Reviewer 2: It should be "Gene←regulates←Gene"

  Shown are six networks from Hetionet [@doi:10.7554/eLife.26726].
  **E.** Common edge-prediction features are correlated with node degree.

>   Shown are five features (Supplemental table {@tbl:edge-prediction}) computed on STRING protein-protein interaction network [@doi:10.1093/nar/gky1131] and the product of the node pair's source and target degrees.

Reviewer 2: Frankly I think a lot of this belongs as the first section of results. there are rather heavy analyses for an introduction.

Author: I agree that the analysis is excessive for an introduction. I want to reference facts established or illustrated by the first figure in the introduction. What is the best way to handle this situation? Can I reference results figures in the introduction?

Reviewer: Just saw this - I would focus the specific examples in the results section exclusively on the work of others. You have a lot of this in the literature that you cite. Then you would move your examination of the networks used in this work to a first subsection of your own results section.

Reviewer 2: I think you mean

> I would focus the specific examples in the ~~results~~introduction section exclusively on the work of others

Reviewer: Yes! @dhimmel is correct 😁

  All five features show a positive relationship with degree, though the magnitude of this correlation is highly variable.

> ](https://github.com/zietzm/xswap-analysis/raw/new-plots/img/fig1.bias_features_combined.png){#fig:degree width="100%"}

Reviewer 2: Best if link to a commit hash rather than branch, since the content pointed to by a branch can CHANGE

### Node degree

Networks contain information about relationships between entities ("edges between nodes").
One key aspect of a node (especially in unweighted networks) is its degree, the number of connections a node makes with other nodes in the network.
Degree is an important metric for differentiating between nodes.
Edge prediction (also known as "link prediction") methods tend to be heavily influenced by degree (Figure {@fig:feature-degree}).
However, reliance on degree may not always be desirable.

> Firstly, degree may be a source of bias in networks.

Reviewer: ```suggestion
Firstly, degree may be influenced by bias in networks.
```
Is the idea that it's a source of bias or that bias influences degree which then leads to biased predictions? There are a lot of difficult questions about degree.

Author: I mean the latter. Degree is an outlet where bias (experimental, inspection, etc.) can manifest in the network. A downstream consequence of this is that predictions can be biased when compared to true connections.

I hesitate to say that degree is influenced by, "bias in networks," as it begs the question, "In what other ways besides degree can bias manifest in networks?" How is something like, "Firstly, bias in the data can distort node degree."

> Secondly, an over-reliance on degree can lead edge prediction methods to make nonspecific or trivial predictions  that fail to identify novel or insightful relationships.

Reviewer: ```suggestion
Secondly, an over-reliance on degree can lead edge prediction methods to make nonspecific or trivial predictions and fail to identify novel or insightful relationships.
```

> We are interested in assessing the effect of node degree on edge prediction methods.

Reviewer: ```suggestion
We sought to understand the effect of node degree on edge prediction methods.
```

How well a network represents the true relationships it attempts to represent depends on a number of factors, especially the methods used to generate the data in the network [@doi:10.1016/j.jprot.2014.01.020; @doi:10.3389/fgene.2015.00260; @doi:10.1038/nbt1116].
We define "degree bias" as the type of misrepresentation that occurs when the fraction of incorrectly existent/nonexistent relationships depends on degree.
Inspection bias indicates that entities are not uniformly studied [@doi:10.1038/nature04209], and it is likely to cause degree bias when networks are constructed using hypothesis-driven findings extracted from the literature, as newly-discovered relationships are not randomly sampled from the set of all true relationships.
For example, the number of publications mentioning a gene has little correlation with its degree in a systematically-derived protein interaction network [Figure 6A in @doi:10.1016/j.cell.2014.10.050].
Therefore, the high correlation between number of publications and degree in low-throughput interaction networks is almost entirely the result of inspection bias.

> In reality, poorly studied genes have just as many interactions as the genes scientists have preferentially examined.

Reviewer: ```suggestion
This evidence suggests that many poorly studied genes have similar numbers of interactions as the genes scientists have preferentially examined.
```

Depending on the type of data being represented, degree biases can also arise due to experimental methods or other factors [@doi:10.1016/j.jprot.2014.01.020].

> Figure {@fig:degree-bias} illustrates the problem of degree bias by showing node degree in separate networks for the same type of data.

Reviewer: This is feeling like it should go in results.

  Systematically-derived and literature-derived networks do not produce linearly correlated node degree.
  Assuming that the inclusion of an edge in the systematic network is independent of node degree, the systematic degree should be linearly correlated with the true number of relationships.
  Edges found in literature-derived networks represent a non-random sample from systematically-derived edges.
  Shown are the same PPI and TF-TG networks as in subfigure A.
](https://github.com/greenelab/xswap-analysis/raw/5a257c9d28e0a4c3cc8f2fd8493f27ff4652f2c6/img/degree_bias.png){#fig:degree-bias width="100%"}

Another reason why a reliance on degree can be unfavorable is that degree-based predictions may be nonspecific.
Nonspecific predictions are based on information that, while potentially associated with true connections, is not reflective of the underlying causal relationships.
For example, predicting that the highest degree node in a network shares edges with the remaining nodes to which it is not connected will often lead to many correct predictions, despite the fact that the predictions were generic to all nodes in the network.

> Gillis et al. [@doi:10.1371/journal.pone.0017258] explored the concept of prediction specificity with respect to gene function prediction, in which predictions based on gene multifunctionality could be, "potentially misleading with respect to causality."

Reviewer: ```suggestion
Gillis et al. [@doi:10.1371/journal.pone.0017258] examine prediction specificity in the context of gene function prediction and found that many predictions appear to rely primarily on multifunctionality and could be "potentially misleading with respect to causality."
```

Author: Agree with the suggestion!

Is past or present tense appropriate for describing previous work by others? The four verbs in this sentence ("examine", "found", "appear", "could be") are not all the same tense.

Edit: Resolved in https://github.com/greenelab/xswap-manuscript/pull/7/commits/e39a2fbfe3df4add6895a95d8168effbd0f6b5b7

> The authors found that, "[N]ode degree is unambiguously linked to multifunctionality," and proposed that, "[T]here are no computational gene function assignments that can be shown to preclude the possibility they reflect only underlying multifunctionality."

Reviewer: ```suggestion
```

> We believe that edge prediction nonspecificity due to node degree imbalance is a general phenomenon across other types of data.

Reviewer: ```suggestion
```

### Node degree

Networks contain information about relationships between entities ("edges between nodes").

> One key property of a node (especially in unweighted networks) is its degree, the number of connections a node makes with other nodes in the network.

Reviewer 2: Maybe "one key aspect" because I think of properties as node attributes, which is a bit distinct from the count of edges.

Degree is an important metric for differentiating between nodes and it is heavily relied-upon by a number of edge prediction (also known as "link prediction") features.
However, reliance on degree may not always be desirable.
Firstly, degree may be a source of bias in networks.

> Secondly, an over-reliance on degree can lead edge prediction methods to make nonspecific or trivial predictions for some tasks, failing to identify the insightful types of relationships desired.

Reviewer 2: ```suggestion
Secondly, an over-reliance on degree can lead edge prediction methods to make nonspecific or trivial predictions that fail at identifying novel or insightful relationships.
```

> Most biomedical data networks are imperfect representations of the true relationships.

Reviewer 2: ```suggestion
Most biomedical data networks are imperfect representations of the true set of relationships.
```
